### PR TITLE
Add ssz spec test

### PR DESCRIPTION
**Pull request by fearlessfe**
Add spec test for type `Fork`
related #3

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -9,8 +9,15 @@ check_nasm:
...
 
+SPEC_VERSION ?= v1.5.0-alpha.8
+
 # Update deps target to include nasm check
 deps: check_nasm
 	@echo "Installing dependencies"
 	@git submodule update --init --recursive
 	@cd bls && make -f Makefile.onelib ETH_CFLAGS=-DBLS_ETH LIB_DIR=lib
+
+	mkdir -p consensus-spec-tests
```
Comment (GrapeBaBa) [inline Makefile:20]: Use a separate the task would be better such as if someone want to build it in local but don't want to do spec tests.

File: build.zig.zon [hunk 2]
```diff
@@ -22,10 +22,7 @@
     // `zig build --fetch` can be used to fetch all dependencies of a package, recursively.
     // Once all dependencies are fetched, `zig build` no longer requires
     // internet connectivity.
-    // .dependencies = .{ .zabi = .{
-    //     .url = "https://github.com/Raiden1411/zabi/archive/79c77aa4a39d41ed50033fd922f4d37c0f0638bb.tar.gz",
-    //     .hash = "122095aaf2d09e3286bf8b00b773b65c1632639de6c5005de881754df7c04efc7e98",
-    // } },
+    .dependencies = .{ .yaml = .{ .url = "git+https://github.com/optimism-java/zig-yaml.git#7a7c5162230bb08f73ec22a063010ed7c76ccded", .hash = "1220a6d7148f7777d997ee663e0efe0fb9c57a38658ea164bcd267e8391c3ce0aa87" } },
```
Comment (GrapeBaBa) [inline build.zig.zon:25]: Why snappy/ssz directly imported but this lib outside, should we unify them？

File: src/spec_tests/ssz_static/root.zig [hunk 3]
```diff
@@ -4,26 +4,123 @@ const ssz = @import("../../ssz/ssz.zig");
 const types = @import("../../consensus/types.zig");
 const snappy = @import("../../snappy/snappy.zig");
 
-test "hash tree root" {
-    const fork = types.Fork{
-        .previous_version = [4]u8{ 0x75, 0xeb, 0x7f, 0x25 },
-        .current_version = [4]u8{ 0x10, 0xd4, 0xe2, 0x7f },
-        .epoch = 8876772290899440384,
-    };
+const Yaml = @import("yaml").Yaml;
```
Comment (GrapeBaBa) [inline src/spec_tests/ssz_static/root.zig:7]: Why upper case here？

File: src/spec_tests/ssz_static/root.zig [hunk 4]
```diff
@@ -4,26 +4,123 @@ const ssz = @import("../../ssz/ssz.zig");
...
+
+        while (try iter.next()) |entry| {
+            if (entry.kind == .directory) {
+                hasSubDir = true;
+                const fullPath = try std.fs.path.join(allocator, &[_][]const u8{ currentPath, entry.name });
+                try list.append(fullPath);
+            }
+        }
+        if (!hasSubDir) {
+            // std.debug.print("currentPath: {s}\n", .{currentPath});
```
Comment (GrapeBaBa) [inline src/spec_tests/ssz_static/root.zig:59]: remove comment out code

File: src/spec_tests/ssz_static/root.zig [hunk 5]
```diff
@@ -4,26 +4,123 @@ const ssz = @import("../../ssz/ssz.zig");
...
-test "hash tree root" {
-    const fork = types.Fork{
-        .previous_version = [4]u8{ 0x75, 0xeb, 0x7f, 0x25 },
-        .current_version = [4]u8{ 0x10, 0xd4, 0xe2, 0x7f },
-        .epoch = 8876772290899440384,
-    };
+const Yaml = @import("yaml").Yaml;
+
+const gpa = testing.allocator;
+
```
Comment (GrapeBaBa) [inline src/spec_tests/ssz_static/root.zig:10]: Add comment for functions

File: Makefile (hunk 1 above)
Comment (fearlessfe) [reply Makefile:21]: done

File: src/spec_tests/ssz_static/root.zig (hunk 5 above)
Comment (fearlessfe) [reply src/spec_tests/ssz_static/root.zig:10]: donw

**Pull request merged**